src/admin/components/ResourceTabs/PolygonReviewTab/components/Polygons.tsx

```diff
@@ -0,0 +1,251 @@
+import Text from "@/components/elements/Text/Text";
+import Icon from "@/components/extensive/Icon/Icon";
+import { IconNames } from "@/components/extensive/Icon/Icon";
+import ModalConfirm from "@/components/extensive/Modal/ModalConfirm";
+import { useModalContext } from "@/context/modal.provider";
+import { useSitePolygonData } from "@/context/sitePolygon.provider";
+import {
+  fetchDeleteV2TerrafundPolygonUuid,
+  fetchGetV2TerrafundGeojsonComplete,
+  fetchGetV2TerrafundPolygonBboxUuid
+} from "@/generated/apiComponents";
+
+import PolygonDrawer from "./PolygonDrawer/PolygonDrawer";
+
+export interface IPolygonItem {
+  id: string;
+  status: "draft" | "submitted" | "approved" | "needs-more-information";
+  label: string;
+  uuid: string;
+}
+
+export interface IpolygonFromMap {
+  isOpen: boolean;
+  uuid: string;
+}
+export interface IPolygonProps {
+  menu: IPolygonItem[];
+  polygonFromMap?: IpolygonFromMap;
+  setPolygonFromMap?: any;
+  refresh?: () => void;
+  mapFunctions: any;
+}
+const statusColor = {
+  draft: "bg-pinkCustom",
+  submitted: "bg-blue",
+  approved: "bg-green",
+  "needs-more-information": "bg-tertiary-600"
+};
+
+export const polygonData = [
+  { id: "1", name: "Site-polygon001.geojson", status: "We are processing your polygon", isUploaded: false },
+  { id: "2", name: "Site-polygon002.geojson", status: "We are processing your polygon", isUploaded: false },
+  { id: "3", name: "Site-polygon003.geojson", status: "We are processing your polygon", isUploaded: true },
+  { id: "4", name: "Site-polygon004.geojson", status: "We are processing your polygon", isUploaded: true },
+  { id: "5", name: "Site-polygon005.geojson", status: "We are processing your polygon", isUploaded: true }
+];
+
+const Polygons = (props: IPolygonProps) => {
+  const [isOpenPolygonDrawer, setIsOpenPolygonDrawer] = useState(false);
+  const [polygonMenu, setPolygonMenu] = useState<IPolygonItem[]>(props.menu);
+  const { polygonFromMap, setPolygonFromMap, mapFunctions } = props;
+  const { map } = mapFunctions;
+  const containerRef = useRef<HTMLDivElement>(null);
+  const { openModal, closeModal } = useModalContext();
+  const [selectedPolygon, setSelectedPolygon] = useState<IPolygonItem>();
+  const [isPolygonStatusOpen, setIsPolygonStatusOpen] = useState(false);
+  const context = useSitePolygonData();
+  const reloadSiteData = context?.reloadSiteData;
+  const { toggleUserDrawing } = context || {};
+
+  useEffect(() => {
+    setPolygonMenu(props.menu);
+  }, [props.menu]);
+  useEffect(() => {
+    if (!isOpenPolygonDrawer) {
+      setSelectedPolygon(undefined);
+    }
+  }, [isOpenPolygonDrawer]);
+
+  useEffect(() => {
+    if (polygonFromMap?.isOpen) {
+      const newSelectedPolygon = polygonMenu.find(polygon => polygon.uuid === polygonFromMap.uuid);
+      setSelectedPolygon(newSelectedPolygon);
+      setIsOpenPolygonDrawer(true);
+    } else {
+      setIsOpenPolygonDrawer(false);
```

Comment:
:art: I haven't read this component carefully enough to be sure, but perhaps the effect above can be dropped if `setSelectedPolygon(undefined)` is added in this block. If that's feasible, it would reduce re-renders for this component since every call to `setState` causes a re-render.

Reply:
Done.